src/processing/resources/page.js

```diff
@@ -118,40 +106,38 @@ class PageProcessor extends ResourceProcessorBase {
 
         var $ = whacko.load(html);
 
-        actualCharset = actualCharset || defaultCharset;
-
-        var rightCharset = this._getRightCharset($, defaultCharset, actualCharset);
+        if (charset.fromMeta(this._getPageMetas($))) {
+            //restart injecting
+            this.processResource(html, ctx, charset, urlReplacer, processingOpts);
```

Comment:
Why we call `this.processResource` here and instantly return `null` which tells caller to restart whole processing?